src/streaming/text/TextTracks.js

```diff
@@ -499,7 +519,24 @@ function TextTracks(config) {
                             }
                             track.manualCueList.push(cue);
                         } else {
-                            if (!_extendLastCue(cue, track)) {
+                            // Handle adjacent cues
+                            let prevCue;
+                            if (track.cues && track.cues.length !== 0) {
+                                prevCue = track.cues[track.cues.length - 1];
+                            }
+
+                            if (_areCuesAdjacent(cue, prevCue)) {
+                                if (!_extendLastCue(cue, prevCue)) {
```

Comment:
Yes, I believe so. Using the [test stream mentioned above](https://broadpeak-tv.github.io/test-content/dash/imsc-image/imsc1_img.mpd) as an example: if `streaming.text.extendSegmentedCues` is set to false, then the adjacent cues at 00:03 and 00:04 are treated as not identical (even though they are).
This means that two cues with identical subtitle text are rendered at 00:03 and 00:04, with different cue id's (rather than extending the first), but with the other code changes, it appears on screen as a continuous (extended) caption and there is no flickering in between the rendering.